spec/models/accounting/qb/transaction_builder_spec.rb

```diff
@@ -1,111 +1,167 @@
-      amount: 100
-    )
-  }
-
-  let(:line_item_2) {
-    build(:line_item,
-      account: bank_account,
-      posting_type: 'Credit',
-      amount: 25
-    )
-  }
-
-  let(:line_item_3) {
-    build(:line_item,
-      account: office_account,
-      posting_type: 'Credit',
-      amount: 75
-    )
-  }
+  let(:memo) { "I am a memo" }
+  let(:description) { "I am a line item description" }
+  let(:date) { Time.zone.today }
 
   subject do
-    described_class.new(instance_double(Division, qb_connection: connection, principal_account: principal_account))
+    described_class.new(instance_double(Division,
+                                        qb_connection: connection, principal_account: principal_account))
   end
 
   before do
     allow(subject).to receive(:class_service).and_return(class_service)
-    allow(subject).to receive(:customer_entity).and_return(customer_entity)
     allow(subject).to receive(:department_reference).and_return(department_reference)
-
-    # since transaction does not exist yet
-    transaction.line_items << [line_item_1, line_item_2, line_item_3]
   end
 
-  let(:qb_customer_id) { '91234' }
-  let(:qb_department_id) { '4012' }
-  let(:customer_entity) { instance_double(Quickbooks::Model::Entity) }
+  let(:qb_customer_id) { "91234" }
+  let(:qb_department_id) { "4012" }
   let(:department_reference) { instance_double(Quickbooks::Model::BaseReference, value: qb_department_id) }
-  let(:customer_name) { 'A cooperative with a name' }
+  let(:customer_name) { "A cooperative with a name" }
   let(:organization) { create(:organization, name: customer_name, qb_id: nil) }
 
-  it 'calls create with correct data' do
-    je = subject.build_for_qb transaction
-
-    expect(je.line_items.count).to eq 3
-    expect(je.private_note).to eq memo
-    expect(je.doc_number).to eq 'MS-Managed'
-    expect(je.txn_date).to be_nil
-
-    list = je.line_items
-    expect(list.map(&:amount)).to eq transaction.line_items.map(&:amount)
-    expect(list.map(&:description).uniq).to eq ['I am a line item description']
-
-    details = list.map { |i| i.journal_entry_line_detail }
-    expect(details.map { |i| i.posting_type }.uniq).to match_array %w(Debit Credit)
-    expect(details.map { |i| i.entity }.uniq).to eq [customer_entity]
-    expect(details.map { |i| i.class_ref.value }.uniq).to eq [loan_id]
-    # parent_ref on a qb class is not queryable in qb api, so not tested here; must use qb
-    # to verify that parent class set correctly
-    expect(details.map { |i| i.department_ref.value }.uniq).to eq [qb_department_id]
-    expect(details.map { |i| i.account_ref.value }.uniq).to match_array [qb_bank_account_id, qb_principal_account_id, qb_office_account_id]
+  describe "a purchase" do
+    before do
+      transaction.line_items << [prin_line_item, line_item_2]
+    end
+    let(:disbursement_type) { :check }
+    let(:check_number) { "123" }
+    let(:transaction) do
+      Accounting::Transaction.create(
```

Comment:
Please use `create!` unless you're dealing with a potential validation failure. (IMO create! should have been the default when they designed Rails!)